posthog/views.py

```diff
@@ -135,3 +138,56 @@ def preflight_check(request: HttpRequest) -> JsonResponse:
         }
 
     return JsonResponse(response)
+
+
+def get_redis_key_type_value_tuple(key: bytes, redis_client):
+    """Get a tuple with a Redis key, type, and value from a Redis key."""
+    redis_key = key.decode("utf-8")
+    redis_type = redis_client.type(redis_key).decode("utf8")
+
+    if redis_type == "string":
+        value = redis_client.get(key)
+
+    elif redis_type == "hash":
+        value = redis_client.hgetall(key)
+
+    elif redis_type == "zset":
+        value = redis_client.zrange(key, 0, -1)
+
+    elif redis_type == "list":
+        value = redis_client.lrange(key, 0, -1)
+
+    elif redis_type == "set":
+        value = redis_client.smembers(key)
+    else:
+        raise ValueError(f"Key {redis_key} has an unsupported type: {redis_type}")
+
+    return (redis_key, redis_type, value)
+
+
+@staff_member_required
+def redis_values_view(request: HttpRequest):
+    """A Django admin view to list Redis key-value pairs."""
+    if request.method != "GET":
+        return HttpResponseNotAllowed(permitted_methods=["GET"])
+
+    query = request.GET.get("q", None)
+    if query == "":
+        query = None
+
+    redis_client = get_client()
+    # TODO: Results are missing pagination.
+    # Depends on the amount of keys; hoping that for starters there won't be enough to justify pagination.
+    _, key_list = redis_client.scan(cursor=0, match=query)
```

Comment:
Fly-by comment :) This would be super useful for debugging caching related issues as well. We do have a lot of keys in there, so usually a simple `SCAN 0` does not come up with results. Passing in a count `SCAN 0 COUNT 100000` does make Redis work harder however and then we get results.

Reply:
I imagine most folks will already have a redis value in mind they wish to debug, and thus would use the `q` parameter to pass a pattern to `MATCH`, and narrow down the view to only those values.

However, I do want to show _something_ even with no `MATCH` or if the pattern is too generic, so I've added some basic cursor/count based pagination for this reason.

Reply:
There are some caveats, like I'm not keeping the history of cursors so there is no way of jumping pages or going to the previous page (although users can always use their browsers to go back).

But, again, I'm working on the assumption that users will narrow down what they are looking for and won't have too scroll through too many pages.

Reply:
Yeah, that's why we are now adding count (and match if provided by the user).